server/src/main/java/org/opensearch/wlm/tracker/QueryGroupResourceUsage.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.wlm.QueryGroupTask;
+import org.opensearch.wlm.ResourceType;
+import org.opensearch.wlm.WorkloadManagementSettings;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.opensearch.wlm.tracker.QueryGroupResourceUsageTrackerService.PROCESSOR_COUNT;
+
+/**
+ * This class is used to track query group level resource usage
+ */
+public abstract class QueryGroupResourceUsage {
+    private double currentUsage;
+
+    /**
+     * getter for value field
+     * @return resource usage value
+     */
+    public double getCurrentUsage() {
+        return currentUsage;
+    }
+
+    public void setCurrentUsage(double currentUsage) {
+        this.currentUsage = currentUsage;
+    }
+
+    public static QueryGroupResourceUsage from(ResourceType resourceType) {
+        if (resourceType == ResourceType.CPU) {
+            return new QueryGroupCpuUsage();
```

Comment:
These should be singleton instances. Don't really see any reason for initializing them repeatedly and adding GC overhead

Reply:
Sure, I will make these singletons

Reply:
Changed this object with a double type.